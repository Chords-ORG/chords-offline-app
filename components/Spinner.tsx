import React from "react";
import { Modal, StyleSheet, View, ActivityIndicator } from "react-native";

export interface SpinnerProps {
  visible: boolean;
}
export default function Spinner(props: SpinnerProps) {
  return (
    <Modal visible={props.visible} transparent>
      <View style={styles.container}>
        <ActivityIndicator size="large" color="#FFFFFF" />
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#00000066",
  },
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 15,
    alignItems: "center",
  },
  openButton: {
    backgroundColor: "#F194FF",
    borderRadius: 20,
    padding: 10,
    elevation: 2,
  },
  textStyle: {
    color: "white",
    fontWeight: "bold",
    textAlign: "center",
  },
  modalText: {
    marginBottom: 15,
    textAlign: "center",
  },
});
