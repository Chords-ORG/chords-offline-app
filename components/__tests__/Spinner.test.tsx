import renderer from 'react-test-renderer';
import React from "react";
import Spinner from "../Spinner";

describe("<Spinner/>", () => {
  it("rendering visible", () => {
    const tree = renderer.create(<Spinner visible={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("rendering invisible", () => {
    const tree = renderer.create(<Spinner visible={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
