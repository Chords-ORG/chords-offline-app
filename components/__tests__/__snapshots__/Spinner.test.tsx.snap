// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> rendering invisible 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;

exports[`<Spinner/> rendering visible 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000066",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#FFFFFF"
      size="large"
    />
  </View>
</Modal>
`;
